src/geo_traits_impl.rs

```diff
@@ -612,7 +618,7 @@ impl From<crate::PolygonZ> for MultiPolygonZ {
                     if let Some(poly) = last_poly.as_mut() {
                         poly.inner.push(points);
                     } else {
-                        panic!("inner ring without a previous outer ring");
+                        return Err(crate::Error::OrphanedInnerRing);
                         // This is the strange (?) case: inner ring without a previous outer ring
```

Comment:
Should we remove this comment?

Also, should we also return this error in the non-geo-traits code paths like 
https://github.com/tmontaigu/shapefile-rs/blob/40847e6190de45b771acc30c2e921ffb58c71367/src/record/polygon.rs#L589

Reply:
> Also, should we also return this error in the non-geo-traits code paths like

After looking at the `From` implementation, I realized it chooses to ignore errors rather than panic (since this is `From`, not `TryFrom`, it cannot return an error). Which is preferable?

https://github.com/tmontaigu/shapefile-rs/blob/40847e6190de45b771acc30c2e921ffb58c71367/src/record/polygon.rs#L586-L594

Reply:
Thanks, so, by "I'm fine," you mean an erroring varitant is preferable? If so, I agree with you. We can add non erroring varinat when we actually find it useful.